caravel/viz.py

```diff
@@ -147,14 +147,36 @@ def get_df(self, query_obj=None):
         self.results = None
 
         # The datasource here can be different backend but the interface is common
+        timestamp_format = None
+        if self.datasource.type == 'table':
+            dttm_cols = self.datasource.dttm_cols
```

Comment:
Assuming a `datasource.get_col` (that would need to be created), can't this whole block be as simple as 
`timestamp_format = datasource.get_col(query_obj['granularity']).python_date_format `